base/src/com/thoughtworks/go/util/SystemEnvironment.java

```diff
@@ -160,6 +160,11 @@
     public static GoSystemProperty<String> GO_LANDING_PAGE = new GoStringSystemProperty("go.landing.page", "/pipelines");
 
     public static GoSystemProperty<Boolean> FETCH_ARTIFACT_AUTO_SUGGEST = new GoBooleanSystemProperty("go.fetch-artifact.auto-suggest", true);
+    public static GoSystemProperty<String> GO_SSL_CONFIG_FILE_PATH = new GoStringSystemProperty("go.ssl.config.path", "/ssl.config");
+    public static GoSystemProperty<String> USER_CONFIGURED_SSL_CONFIG_FILE_PATH = new GoStringSystemProperty("user.ssl.config.path", "user.ssl.config");
+    public static GoSystemProperty<String> TRANSPORT_PROTOCOL_TO_BE_USED_BY_AGENT = new GoStringSystemProperty("agent.transport.protocol", "TLSv1.2");
```

Comment:
SSLv3 is no longer considered secure, but kept for backwards compatibility with older clients.

@jyotisingh - do we know what is set of user agents (browsers) that are supported by the current transport protocol and ciphers?

A good way to check would be to spin up a server on ec2 and point ssllabs at port 8154. Also ssllabs mentions a bunch of browser capabilities here - https://www.ssllabs.com/ssltest/clients.html